quinn-proto/src/crypto/rustls.rs

```diff
@@ -204,7 +206,7 @@ const RETRY_INTEGRITY_NONCE_V1: [u8; 12] = [
     0x46, 0x15, 0x99, 0xd3, 0x5d, 0x63, 0x2b, 0xf2, 0x23, 0x98, 0x25, 0xbb,
 ];
 
-impl crypto::HeaderKey for HeaderProtectionKey {
+impl crypto::HeaderKey for Box<dyn HeaderProtectionKey> {
```

Comment:
Is there a way we could avoid double-boxing here?

Reply:
I haven't figured out a way. I'm afraid it might be pretty hard and there are still other open issues.

Reply:
Yeah, I think it's impossible, short of reimplementing a good chunk of rustls's QUIC logic to undo rustls's backend abstraction. Not a huge deal.